src/components/experiment/Experiment.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Toast from 'react-bootstrap/Toast';
+import Container from 'react-bootstrap/Container';
+import ExperimentItem from './ExperimentItem';
+
+const generateItems = (experimentUrls: Array<string>) => {
+  const items: Array<JSX.Element> = [];
+  experimentUrls.forEach(url => {
+    items.push(<ExperimentItem url={url} key={url} />);
+  });
+
+  return items;
+};
+
+const Experiment: React.FC = () => {
+  const [hasError, setErrors] = useState(false);
+  const [experimentUrls, setExperimentUrls] = useState([]);
+  const [isUploading, setUploading] = useState(false);
+  const [showToast, setShowToast] = useState(false);
+  const [responseMessage, setResponseMessage] = useState('');
```

Comment:
Do we need both showToast and responseMessage? If the response message is null, that's enough info to know not to show the toast. Conversely, if the responseMessage exists we should show a toast. With that in mind, is the showToast state necessary?

Reply:
You can't directly assign null to a state, typescript won't allow it. I could likely get around this by passing it { message: null } but that feels less than clean. Up to you though, if that makes more sense I'll change it

Reply:
It's fine as is